File: src/main/java/com/lucas/matches/service/util/LicenseParser.java
package com.lucas.matches.service.util;

import com.lucas.matches.service.exception.InvalidLicenseException;
import com.lucas.matches.service.domain.License;
import com.lucas.matches.service.domain.LicenseType;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class LicenseParser {
    private final Logger logger;

    public LicenseParser() {
        this.logger = LoggerFactory.getLogger(LicenseParser.class);
    }

    public List<License> parse(List<String> licenseStringList){
        List<License> licenses = new ArrayList<>();
        licenseStringList.forEach(licenseString -> licenses.add(parse(licenseString)));
        return  licenses;
    }
    private License parse(String licenseString){
        String[] licenseArray = licenseString.split("-");

       if(licenseArray.length < 2){
           logger.error("Invalid license: " + licenseString);
           throw new InvalidLicenseException("Invalid license.");
       }
        License license = new License(LicenseType.value(licenseArray[0]), licenseArray[1]);
        if(license.getLicenseType() == null || license.getId() == null || license.getId().isEmpty()){
            logger.error("Invalid license: " + licenseString);
            throw new InvalidLicenseException("Invalid license.");
        }
        return license;
    }
}

File: src/main/kotlin/com/lucas/matches/service/util/LicenseParser.kt
package com.lucas.matches.service.util

import com.lucas.matches.service.domain.License
import com.lucas.matches.service.domain.LicenseType
import com.lucas.matches.service.exception.InvalidLicenseException
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component
import java.util.function.Consumer

@Component
class LicenseParser {
    private val logger: Logger = LoggerFactory.getLogger(LicenseParser::class.java)
    fun parse(licenseStringList: List<String>): List<License> {
        return licenseStringList.mapNotNull { parse(it)}
    }

    private fun parse(licenseString: String): License? {
        val licenseArray = licenseString.split("-").toTypedArray()
        if (licenseArray.size < 2) {
            logger.error("Invalid license: $licenseString")
            throw InvalidLicenseException("Invalid license.")
        }
        val licenseType = LicenseType.value(licenseArray[0])

        if(licenseType == null){
            logger.error("Invalid license: $licenseString")
            throw InvalidLicenseException("Invalid license.")
        }
        return  License(licenseType, licenseArray[1])

    }

}